jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/editor/S3TreeTable.kt

```diff
@@ -118,20 +117,22 @@ class S3TreeTable(
         launch {
             try {
                 bucket.download(project, objectNode.key, objectNode.versionId, fileWrapper.file.outputStream())
-                withContext(edt) {
-                    // If the file type is not associated, prompt user to associate. Returns null on cancel
-                    fileWrapper.virtualFile?.let {
-                        ApplicationManager.getApplication().runWriteAction {
-                            it.isWritable = false
-                        }
-                        FileTypeChooser.getKnownFileTypeOrAssociate(it, project) ?: return@withContext
-                        // set virtual file to read only
-                        FileEditorManager.getInstance(project).openFile(it, true, true).ifEmpty {
-                            notifyError(project = project, content = message("s3.open.viewer.failed.unsupported"))
+                if (S3VirtualBucket.bucketExists) {
```

Comment:
In general, avoid large if blocks with no else/nothing else after. This would read better inverted as:
```kt
if (!S3VirtualBucket.bucketExists) {
    return
}
```
We should probably also add a log and a `S3Telemetry.downloadObject(project, cancelled)